svc/env.go

```diff
@@ -87,15 +88,34 @@ func InitTracer() func() {
 //
 // Env Vars:
 // * STATSD_ADDR - The host:port of the statsd server.
-func InitMetrics() func() {
-	if addr := os.Getenv("STATSD_ADDR"); addr != "" {
-		metrics.SetEmpireDefaultTags()
-		metrics.Reporter, _ = metrics.NewDataDogMetricsReporter(addr)
+func InitMetrics() (fn func()) {
+	fn = func() {
+		metrics.Close()
 	}
 
-	return func() {
-		metrics.Close()
+	addr := os.Getenv("STATSD_ADDR")
+	if addr == "" {
+		return
 	}
+
+	host, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		return
+	}
+
+	addrs, err := net.LookupHost(host)
+	if err != nil {
+		return
+	}
+
+	if len(addrs) == 0 {
+		return
+	}
+
+	metrics.SetEmpireDefaultTags()
+	metrics.Reporter, _ = metrics.NewDataDogMetricsReporter(net.JoinHostPort(addrs[0], port))
```

Comment:
This seems a little odd to have to do; is it indicative that something downstream stream/within our metrics report/datadog's report that they're not keeping an http client connection/open/something?